conjure/src/test/java/com/palantir/conjure/cli/ConjureCliTest.java

```diff
@@ -166,6 +166,25 @@ public void generatesCleanError_map() {
         assertThat(outputFile.isFile()).isFalse();
     }
 
+    @Test
+    public void generatesCleanError_unique_name() {
+        String[] args = {"compile", "src/test/resources/unique-name-error", outputFile.getAbsolutePath()};
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter printWriter = new PrintWriter(stringWriter);
+        ConjureCli.prepareCommand().setErr(printWriter).execute(args);
+        printWriter.flush();
+        assertThat(stringWriter.toString())
+                .isEqualTo("Type, error, and service names must be unique across locally defined and "
+                        + "imported types/errors:\n"
+                        + "Found duplicate name: test.api.ConflictingName\n"
+                        + "Known names:\n"
+                        + " - test.api.UniqueName\n"
+                        + " - test.api.UniqueName2\n"
+                        + " - test.api.ConflictingName\n");
+        assertThat(outputFile.isFile()).isFalse();
```

Comment:
Huh, we should add some automatic assertj refactors for this:
```suggestion
        assertThat(outputFile).doesNotExist();
```